src/controllers/Flashcards.js

```diff
@@ -0,0 +1,99 @@
+const Flashcard = require('../models/Flashcard');
+const { StatusCodes } = require('http-status-codes');
+
+const getAllFlashcards = async (req, res) => {
```

Comment:
[Optional]
No need to fix this right away if you don't want to. Just add another card in JIRA for us to fix it eventually.

[Pagination]
This works if and only if you have a small number of flashcards. If for instance we had 100,000,000 flashcards in here, you'd try to get all of those and the system would choke cause its just too much data.
Generally, an API like this offers pagination (I think we even did some examples in class). So you'll have query string parameters like `limit` (or `size`) and `offset` (or `start`). The `limit` tells the API the max number of flashcards to return and the `offset` tells it where to start.
Example:
- Given you have 100,000 flashcards
- When a user requests `GET /api/v1/flashcards?limit=100&offset=500`
- Then that user will get flashcards 500-599 inclusive

We also generally put limitations on the max you can ask for with 1,000 or 10,000 being pretty common. So if soomeone asks for 1,000,000 we'd give a 400 error and tell them they can't ask for more than 1,000.
Example:
- Given you have 100,000 flashcards
- When a user requests `GET /api/v1/flashcards?limit=100000&offset=0`
- Then we return `400 Bad Request {'msg':'limit cannot be greater than 1,000'}`

[Access Control & Privacy]
We didn't bake in a way for users to specify whether or not they were willing to share their flashcards.
It would be good to add a new property on the flashcard called "availability" with an enum value of either "Public" or "Private'. Then when we get all flashcards we would filter to just the ones that are marked Public.
We can in theory get way more complicated on this, but "I do/do not want to share this" is a good start.

Reply:
added pagination as recommended